python/sglang/srt/model_executor/cuda_graph_runner.py

```diff
@@ -591,33 +895,82 @@ def capture_one_batch_size(self, bs: int, forward: Callable):
             encoder_lens=encoder_lens,
             return_logprob=False,
             positions=positions,
-            global_num_tokens_gpu=self.global_num_tokens_gpu,
-            global_num_tokens_for_logprob_gpu=self.global_num_tokens_for_logprob_gpu,
-            dp_padding_mode=DpPaddingMode.get_default_mode_in_cuda_graph(),
             global_dp_buffer_len=global_dp_buffer_len,
+            global_num_tokens_gpu=(
+                self.prefill_global_num_tokens_gpu
+                if hasattr(self, "capture_forward_mode")
+                and self.capture_forward_mode == ForwardMode.EXTEND
+                else self.global_num_tokens_gpu
+            ),
+            global_num_tokens_for_logprob_gpu=(
+                self.prefill_global_num_tokens_for_logprob_gpu
+                if hasattr(self, "capture_forward_mode")
+                and self.capture_forward_mode == ForwardMode.EXTEND
+                else self.global_num_tokens_for_logprob_gpu
```

Comment:
another question: do you think it is better if we have one more GraphRunner tailored for prefill? Let's say, one way maybe BaseGraphRunner, DecodeGraphRunner, PrefillGraphRunner. Another way maybe, one class (GraphRunner) and two objects (prefill_runner = GraphRunner(); decode_runner = GraphRunner()).

also cc @merrymercy, since he (and another person iirc) is working on eagle cuda graph refactoring so I dnk whether will be related